docs/store.js

```diff
@@ -230,6 +294,26 @@ var firebaseCalendarStorage = function(params) {
             })
         })
     };
+    
+    self.initializeCalendar = function(calendarObj) {
+        // Initializes a calendar in the database. Adds the calendar creator
+        // as a writer and a reader, sets the calendar as the currentActiveCalendar
+        // and then runs the save method (store cal state and allCalendarIds
+        
+        self.startWork();
+        var addWriterP = self.addWriter(self.user, calendarObj)
+        .then(function() {
+            self.addReader(self.user, calendarObj);
+            self.setActiveById(calendarObj.state.uniqueId);
+            self.save(calendarObj);
+            self.endWork();
```

Comment:
You're not done with work until those other promises return.  You need a `Promise.all` here.